pkg/client/http/http.go

```diff
@@ -99,18 +100,49 @@ func (dc *Client) Call(r *client.Request) (resp client.Response, err error) {
 
 	urlStr := r.API.IntegrationRequest.HTTPBackendConfig.Protocol + "://" + r.API.IntegrationRequest.HTTPBackendConfig.TargetURL
 	httpClient := &http.Client{Timeout: 5 * time.Second}
-	request := r.IngressRequest.Clone(context.Background())
-	request.URL, _ = url.ParseRequestURI(urlStr)
-	//TODO header replace, url rewrite....
+	request := r.IngressRequest.Clone(r.Context)
+	//Map the origin paramters to backend parameters according to the API configure
+	transformedParams, err := dc.MapParams(r)
+	if err != nil {
+		return *client.EmptyResponse, err
+	}
+	params, _ := transformedParams.(*requestParams)
+	request.Body = params.Body
+	request.Header = params.Header
 
+	urlStr = strings.TrimRight(urlStr, "/") + "?" + params.Query.Encode()
+	request.URL, _ = url.ParseRequestURI(urlStr)
 	tmpRet, err := httpClient.Do(request)
 	ret := client.Response{Data: tmpRet}
 	return ret, err
 }
 
-// MappingParams param mapping to api.
-func (dc *Client) MappingParams(req *client.Request) (types []string, reqData []interface{}, err error) {
-	return nil, nil, nil
+// MapParams param mapping to api.
+func (dc *Client) MapParams(req *client.Request) (reqData interface{}, err error) {
+	mp := req.API.IntegrationRequest.MappingParams
+	r := newRequestParams()
+	if len(mp) == 0 {
+		r.Body = req.IngressRequest.Body
+		r.Header = req.IngressRequest.Header.Clone()
+		queryValues, err := url.ParseQuery(req.IngressRequest.URL.RawQuery)
+		if err != nil {
+			return nil, errors.New("Retrieve request query parameters failed")
```

Comment:
I think we need create err together `errors.New()` in next step.

Reply:
agree.